var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -3,160 +3,164 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import os
-
 from spack.package import *
 
 
-class Dftbplus(MakefilePackage):
+class Dftbplus(CMakePackage):
     """DFTB+ is an implementation of the
     Density Functional based Tight Binding (DFTB) method,
     containing many extensions to the original method."""
 
     homepage = "https://www.dftbplus.org"
-    url = "https://github.com/dftbplus/dftbplus/archive/19.1.tar.gz"
-
+    url = "https://github.com/dftbplus/dftbplus/archive/22.1.tar.gz"
+    git = "https://github.com/dftbplus/dftbplus.git"
+
+    maintainers = ["HaoZeke", "aradi"]
+    generator = "Ninja"
+
+    version("main", branch="main")
+    version("22.1", sha256="f0fc9a076aa2d7be03c31a3a845d8151fc0cc0b1d421e11c37044f78a42abb33")
+    version("21.2", sha256="e73aa698ff951b59f7fe2ea027b292bae16abf545c9fdbb11b5b5127f04a3c10")
+    version("21.1", sha256="31d5a488843a05d8589a375307a2832c1fc938f9f7d830c45a062726659e7b0a")
+    version("20.2.1", sha256="6b1827a45b20d1757119a75abcb851cd4362e7abc58601094d029ed5922d6da7")
+    version("20.2", sha256="6b1f10c7efbdaf59acf64d8fb8afc80460c0fbc6c4dbe256b760223cbd28ed2c")
+    version("20.1", sha256="a155ca927c804234587c61c4938d154f31578c816b0ce20eaee3b5d7e39d91dc")
     version("19.1", sha256="4d07f5c6102f06999d8cfdb1d17f5b59f9f2b804697f14b3bc562e3ea094b8a8")
 
-    resource(
-        name="slakos",
-        url="https://github.com/dftbplus/testparams/archive/dftbplus-18.2.tar.gz",
-        sha256="bd191b3d240c1a81a8754a365e53a78b581fc92eb074dd5beb8b56a669a8d3d1",
-        destination="external/slakos",
-        when="@18.2:",
-    )
+    depends_on("cmake@3.16:", type="build")
+    depends_on("ninja", type="build")
 
-    variant("mpi", default=True, description="Build an MPI-paralelised version of the code.")
+    depends_on("blas", when="-mpi")
+    depends_on("lapack", when="-mpi")
+
+    depends_on("arpack-ng", when="+arpack")
+    depends_on("simple-dftd3", when="+sdftd3")
+    depends_on("elsi", when="+elsi")
+    depends_on("magma", when="+gpu")
+    depends_on("mpi", when="+mpi")
+    depends_on("plumed", when="+plumed")
+    depends_on("scalapack", when="+mpi")
+    depends_on("python", when="+python")
 
     variant(
-        "gpu",
-        default=False,
-        description="Use the MAGMA library " "for GPU accelerated computation",
+        "build_type",
```

Comment:
FYI. Usually `variant` directives appear before `depends_on` directives to show the build options being exposed to the install process before defining dependencies, which may be depend on those variants.

I'm curious as to the motivation for putting the dependencies first.

Reply:
Ah, my bad, I didn't notice the convention but that makes sense. The motivation as such was just that the `variant` directives were visually more noisy. Will update.